src/main/scala/nl/knaw/dans/easy/fedora2vault/SimpleChecker.scala

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (C) 2020 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.fedora2vault
+
+import nl.knaw.dans.common.lang.dataset.AccessCategory.{ OPEN_ACCESS, REQUEST_PERMISSION }
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import nl.knaw.dans.pf.language.emd.EasyMetadataImpl
+
+import scala.util.{ Failure, Success, Try }
+import scala.xml.Node
+
+case class SimpleChecker(bagIndex: BagIndex) extends DebugEnhancedLogging {
+
+  def isSimple(emd: EasyMetadataImpl, ddm: Node, amd: Node, jumpOff: Seq[String]): Try[Unit] = {
+    val doi = emd.getEmdIdentifier.getDansManagedDoi
+    val triedMaybeVaultResponse = bagIndex.bagByDoi(doi).map(_.toSeq)
+    val violations = Seq(
+      "1: DANS DOI" -> (if (Option(doi).isEmpty) Seq("not found")
+                               else Seq[String]()),
+      "2: has jump off" -> jumpOff,
+      "3: invalid title" -> Option(emd.getEmdTitle.getPreferredTitle)
+        .filter(_.toLowerCase.contains("thematische collectie")).toSeq,
+      "4: invalid rights" -> findInvalidRights(emd),
+      "5: invalid state" -> findInvalidState(amd),
+      "6: DANS relations" -> findDansRelations(ddm),
+      "7: is in the vault" -> triedMaybeVaultResponse.getOrElse(Seq("IO exception")),
+    ).filter(_._2.nonEmpty).toMap
+
+    violations.foreach { case (rule, violations) =>
+      violations.foreach(s => mockFriendlyWarn(s"violated $rule $s"))
+    }
+    lazy val errorMessage: String = violations.keys
+      .map(_.replaceAll(":.*", ""))
+      .mkString("Not a simple dataset. Violates rule ", ", ", "")
+    for {
+      _ <- triedMaybeVaultResponse // an IOException is not a violation
+      _ <- if (violations.isEmpty) Success(())
+           else Failure(new Exception(errorMessage))
+    } yield ()
+  }
+
+  /** An interpolated string is a method. It needs evaluation before passing in to define expectations. */
```

Comment:
Does someone else see options on https://scalamock.org/user-guide/advanced_topics/ for a solution that doesn't influence the production code?